app/models/on_duty_locator.rb

```diff
@@ -0,0 +1,22 @@
+class OnDutyLocator
+  def initialize(time, rota_slots)
+    @time = time
+    @rota_slots = rota_slots
+  end
+
+  def locate
+    on_duty_organisation_uid
+  end
+
+  private
+
+  attr_reader :time, :rota_slots
+
+  def slot_with_on_duty_shift
+    rota_slots.detect { |slot| slot.shift.starting_time < time }
```

Comment:
It is. The goal is to get to a something like a `ShiftSequenceForRota`, so we can more actually select the slot/shift. There is a fair amount of work to be done in that regard as, for instance, shifts can currently have no ending time, or representation of that concept. Without that it becomes difficult to deal with where one shift ends and another starts when retrieving based on a timestamp.